tests/integration/test_random_creation.py

```diff
@@ -33,6 +39,447 @@ def test_randn():
+
+
+@pytest.mark.xfail(
+    EAGER_TEST,
+    reason="cuNumeric does not respect seed in Eager mode",
+)
+def test_default_rng_bitgenerator():
+    seed = 12345
+    rng_np_1 = np.random.default_rng(np.random.PCG64(seed))
+    rng_num_1 = num.random.default_rng(num.random.XORWOW(seed))
+    rng_np_2 = np.random.default_rng(np.random.PCG64(seed))
+    rng_num_2 = num.random.default_rng(num.random.XORWOW(seed))
+    assert rng_np_1.random() == rng_np_2.random()
+    assert rng_num_1.random() == rng_num_2.random()
+
+
+@pytest.mark.xfail(
+    EAGER_TEST,
+    reason="cuNumeric does not respect seed in Eager mode",
+)
+def test_default_rng_generator():
+    steps = 3
+    seed = 12345
+    seed_rng_np, seed_rng_num = gen_random_from_both("default_rng", seed)
+    np_rngs = [np.random.default_rng(seed_rng_np) for _ in range(steps)]
+    num_rngs = [num.random.default_rng(seed_rng_num) for _ in range(steps)]
+    np_seq_1 = [rng.random() for rng in np_rngs]
+    num_seq_1 = [rng.random() for rng in num_rngs]
+
+    rng_np, rng_num = gen_random_from_both("default_rng", seed)
+    np_seq_2 = [rng_np.random() for _ in range(steps)]
+    num_seq_2 = [rng_num.random() for _ in range(steps)]
+    assert np_seq_1 == np_seq_2
+    assert num_seq_1 == num_seq_2
+
+
+@pytest.mark.parametrize("shape", [(20, 2, 31), (10000,)], ids=str)
+def test_rand(shape):
+    arr_np, arr_num = gen_random_from_both("rand", *shape)
+    assert arr_np.shape == arr_num.shape
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+
+
+LOW_HIGH = [
+    (5, 10000),
+    (-10000, 5),
+]
+SMALL_RNG_SIZES = [5, 1024, (1, 2)]
+LARGE_RNG_SIZES = [10000, (20, 50, 4)]
+ALL_RNG_SIZES = SMALL_RNG_SIZES + LARGE_RNG_SIZES
+INT_DTYPES = [np.int64, np.int32, np.int16]
+UINT_DTYPES = [np.uint64, np.uint16, np.uint0]
+
+
+@pytest.mark.parametrize("size", ALL_RNG_SIZES, ids=str)
+@pytest.mark.parametrize("low, high", LOW_HIGH, ids=str)
+@pytest.mark.parametrize("dtype", INT_DTYPES, ids=str)
+def test_randint_basic_stats(low, high, size, dtype):
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low=low, high=high, size=size, dtype=dtype
+    )
+    assert arr_np.dtype == arr_np.dtype
+    assert arr_np.shape == arr_num.shape
+    assert np.min(arr_num) >= low
+    assert np.max(arr_num) <= high
+
+
+@pytest.mark.parametrize("low", [1024, 1025, 12345], ids=str)
+@pytest.mark.parametrize("size", LARGE_RNG_SIZES, ids=str)
+@pytest.mark.parametrize("dtype", INT_DTYPES, ids=str)
+def test_randint_low_limit_only(low, size, dtype):
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low, size=size, dtype=dtype
+    )
+    assert arr_np.dtype == arr_np.dtype
+    assert arr_np.shape == arr_num.shape
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+
+
+@pytest.mark.xfail(
+    EAGER_TEST,
+    reason="cuNumeric randint high limit is inclusive in Eager mode",
```

Comment:
@manopapad this seems bizarre, right? Eager mode should just falling back to numpy so how are things disagreeing with numpy

Reply:
:+1: @yimoj let's get #966 merged first with fixes then update this PR for working tests to avoid multiple PRs